### PR TITLE
Added an additional GET parameter to profile sandbox requests.

### PR DESCRIPTION
**Pull request by zinigor**
The additional constant enables adding an XDEBUG_PROFILE constant to all sandbox requests to allow for easier profiling.

#### Changes proposed in this Pull Request:
* An additional constant added that sends an extra GET parameter with all sandbox requests.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No tracking changes.

#### Testing instructions:
* Sandbox your Jetpack installation using the `JETPACK__SANDBOX_DOMAIN` constant.
* Enable Xdebug on your sandbox using `sudo xdebug enable`.
* Add a filter to enable the parameter addition
```
add_filter( 'jetpack_sandbox_add_profile_parameter', '__return_true' ); 
```
* The code above will add the parameter to all requests. You can parse the URL and HOST provided to enable it only for certain requests
* Create a post using a CLI command:
```
wp post generate --count=1
```
* Observe the sync request in the output log. Note that the URL listed in the output will be the original one, not the sandboxed one, so it will not have the profile parameter added.
* Observe a cachegrind output file in your sandbox `/tmp` folder. Be careful, these can be quite large.

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

File: _inc/jetpack-server-sandbox.php [hunk 1]
```diff
@@ -18,17 +18,29 @@ function jetpack_server_sandbox_request_parameters( $sandbox, $url, $headers ) {
...
+		case 'dashboard.wordpress.com':
+			$host = isset( $headers['Host'] ) ? $headers['Host'] : $url_host;
+			$url  = preg_replace(
+				'@^(https?://)' . preg_quote( $url_host, '@' ) . '(?=[/?#].*|$)@',
+				'${1}' . $sandbox,
+				$url,
+				1
+			);
+
+			l( 'WEHERE' );
```
Comment (jeherve) [inline _inc/jetpack-server-sandbox.php:33]:
```suggestion
```

**Review by jeherve — DISMISSED** [review]
Do you think you could add some documentation explaining how this could be used? Maybe here? 
https://github.com/Automattic/jetpack/blob/95b9ac43f162f85adc396f45c59035ed7f0525b8/docker/README.md#L435

**Comment by leogermani** [comment]
> Do you think you could add some documentation explaining how this could be used? Maybe here?
> 
> https://github.com/Automattic/jetpack/blob/95b9ac43f162f85adc396f45c59035ed7f0525b8/docker/README.md#L435

Added

**Review by vitozev — COMMENTED** [review]
I did some functional tests and everything is working as expected, all the requests under monitoring went through my sandbox, and `cachegrind.out.*` profiling files were generated. Nice job!

**Review by kraftbj — DISMISSED** [review]

**Review by kraftbj — DISMISSED** [review]

**Review by kraftbj — APPROVED** [review]

**Pull request merged**